### PR TITLE
MSM/MVLookup: add challenges into column environment

### PR DESCRIPTION
**Pull request by dannywillems**
The generation of the challenges should be at the same round for both the prover and the verifier.
We do not have a typed sponge unfortunately to detect that we coin at the right time. Maybe later.
I am integrating lookups for the serialization subcircuits. We will have tests when this is done.

### CODE REVIEW

File: msm/src/prover.rs [hunk 1]
```diff
@@ -152,23 +152,37 @@ where
...
-        domain,
+    let column_env = {
+        let challenges = if let Some(lookup_env) = lookup_env.as_ref() {
+            Challenges {
+                alpha,
+                // NB: as there is on permutation argument, we do use the beta
+                // field instead of a new one for the evaluation point.
+                beta: lookup_env.beta,
+                gamma: G::ScalarField::zero(),
+                joint_combiner: Some(lookup_env.joint_combiner),
```
Comment (volhovm) [inline msm/src/prover.rs:163]: Suggestion: keeping just one set of Challenges but doing:
`lookup_env.map(|x| x.joint_combiner)` 
or
`lookup_env.as_ref().map(|x| x.joint_combiner)`

and same with `beta`

**Review by volhovm — APPROVED** [review]

File: msm/src/prover.rs (hunk 1 above)
Comment (dannywillems) [reply msm/src/prover.rs:163]: Can not be used for beta, as it is not a map on an option

Comment (volhovm) [reply msm/src/prover.rs:163]: Ah yes indeed. Smth like `lookup_env.map(|x| x.beta).unwrap_or(F::zero())`?

Comment (dannywillems) [reply msm/src/prover.rs]: Done in https://github.com/o1-labs/proof-systems/pull/1933/commits/095d2b23a00a4e3d5260b0eb3038fc77d8d8de7f

**Pull request merged**